gflownet/envs/base.py

```diff
@@ -207,19 +218,51 @@ def step(self, action: Tuple[int]) -> Tuple[List[int], Tuple[int], bool]:
             False, if the action is not allowed for the current state, e.g. stop at the
             root state
         """
-        # If env is done, return invalid
-        if self.done:
-            return self.state, action, False
         # If action not found in action space raise an error
         if action not in self.action_space:
             raise ValueError(
                 f"Tried to execute action {action} not present in action space."
             )
-        action_idx = self.action_space.index(action)
+        # If env is done, return invalid
+        if self.done:
+            return False, self.state, action, False
         # If action is in invalid mask, exit immediately
-        if self.get_mask_invalid_actions_forward()[action_idx]:
-            return self.state, action, False
-        return None, None, None
+        if not (self.skip_mask_check or skip_mask_check):
+            action_idx = self.action_space.index(action)
+            if self.get_mask_invalid_actions_forward()[action_idx]:
+                return False, self.state, action, False
+        return True, self.state, action, True
+
+    @abstractmethod
+    def step(
+        self, action: Tuple[int], skip_mask_check: bool = False
+    ) -> Tuple[List[int], Tuple[int], bool]:
+        """
+        Executes step given an action.
+
+        Args
+        ----
+        action : tuple
+            Action from the action space.
+
+        skip_mask_check : bool
```

Comment:
The argument `skip_mask_check` has been added to the docstring here but many of the other environments do not have it in the docstring of their own `step()` function.

Reply:
Thanks for spotting this!

Reply:
Done